### ISSUE TITLE
Renaming eWoms.

### ISSUE THREAD

**Issue by dr-robertk** [OPM/opm-models#531]
For some time now we have been wanting to rename the opm module ewoms to something more meaningful. Please add suggestions for new module names so that we can vote on this. 
My suggestions is: 

1. opm-models

**Comment by joakim-hove** [comment]
My suggestion is merge with opm-simulators

**Comment by dr-robertk** [comment]
@joakim-hove: Quick reply that was ;-)

**Comment by andlaus** [comment]
Note: because somebody else needs to do this work, the following should be interpreted as my 2c, but I don't really care if this sways the decision into any direction.

IMO the problem here is that ewoms is a jack-of-all-trades module, so the name "ewoms" is not too inappropriate. The best name for how is used by the OPM downstream code (i.e., `ebos` and its derivates) is essentially

opm-numerics

Further, note that taking @joakim-hove's suggestion to its logical conclusion would come down to merging all modules together, including opm-flowdiagnostics and opm-upscaling. I continue to have doubts that this is a good idea if this code base is supposed to be useful for anything else than ECL-surrogate reservoir simulators. (In this case the project should be renamed. Even then, I have severe doubts if this a good idea from the technical POV.)

Finally, isn't it more pressing to clean up the slack w.r.t. the big opm-common, opm-parser, opm-output, opm-core and opm-autodiff reshuffling before adding even more confusion and also, why is the I/O code for a specific file format called "common"?

**Comment by atgeirr** [comment]
I can go with either suggestion above (opm-models, opm-numerics or combining with opm-simulators) I prefer calling it opm-models and doing a rename at this point rather than combining the repos, unless it can be done in good time before the next release.

**Comment by alfbr** [comment]
> Further, note that taking @joakim-hove's suggestion to its logical conclusion would come down to merging all modules together, including opm-flowdiagnostics and opm-upscaling.

I belive this is a misunderstanding. I don't think anybody advocates merging all modules together. ewoms may very well be a separate repository, but it is now in the build chain of opm-simulators, and that is unfortunate.

> Finally, isn't it more pressing to clean up the slack w.r.t. the big opm-common, opm-parser, opm-output, opm-core and opm-autodiff reshuffling before adding even more confusion

Please be specific, what clean-up are you suggesting?

@dr-robertk , back to the orginal question. I side with @joakim-hove here, I would rather see time used to unify ewoms and opm-simulators. I am not sure a rename beforehand makes much sense if we are to unify them, e.g., right after the October release. It may or may not make sense to keep ewoms as a separate repository, but I do believe the current dependency situation is unfortunate.

**Comment by atgeirr** [comment]
> Finally, isn't it more pressing to clean up the slack w.r.t. the big opm-common, opm-parser, opm-output, opm-core and opm-autodiff reshuffling before adding even more confusion and also, why is the I/O code for a specific file format called "common"?

I think we have cleaned up significant parts of this already. There are still some files in the opm/core subdir of opm-simulators, but they are only waiting there to see if they will be obsoleted (and removed) or refurbished (and moved to e.g. opm/simulators/wells).

**Comment by GitPaean** [comment]
Not sure about the possible side effects and technical difficulties, and related issues.

If we were about to merge ewoms to opm-simulators, I would suggest to only merge the parts of ewoms that needs to be in the compilation chain of opm-simulators.  ewoms can still be a separate repo on its own to have its extension and good features.

**Comment by totto82** [comment]
All options have good points. But I think renaming ewoms to opm-models is the best solution. Not perfect, since it contains other stuff like linear solvers etc, but good enough. A too long but maybe more describing option could be opm-modelingframework or opm-simulationframework. Or just opm-framework.

**Comment by andlaus** [comment]
> But it is now in the build chain of opm-simulators, and that is unfortunate.

so is `opm-common`, `opm-grid` and `opm-material`. also, `libecl` is in the same build chain and -- in contrast to the proper OPM modules -- constantly produced headaches for me. Further, if you do not just care about the current version of `flow` (for example, I've heard that there are people around who once only cared about what's now `flow-legacy`), you would also need to merge `opm-legacy` and `opm-upscaling` at least.

> Please be specific, what clean-up are you suggesting?

basically, putting the files to their right locations: e.g., `opm/core` is present in `opm-upscaling` and `opm/simulators` and all modules except opm-grid and opm-material have headers located under directories that are not following the `opm/$module` convention. the worst offender is `opm-common` which has the directories `common`, `io`, `json`, `output` and `parser` as sub-directory of `opm`. good luck to newbie developers with locating a file quickly.

**Comment by dr-robertk** [comment]
This issue is about finding a better name for ewoms. I would like to ask everybody to keep the comments focused on this topic or continue the discussion in a different thread somewhere else.

**Comment by dr-robertk** [comment]
From the above (sometimes lengthy) comments I conclude the following from the 7 votes above: 

1. Renaming to opm-models (3 votes)
2. Merging with opm-simulators (2 votes)
3. Renaming to opm-numerics (1 vote) 

@GitPaean vote is a bit unclear and most likely fits into 2.

**Comment by joakim-hove** [comment]
If the merge with opm-simulators does not reach up `opm-models` will get my secondary vote.

**Comment by andlaus** [comment]
For the record: I'm also fine with `opm-models`

**Comment by atgeirr** [comment]
We seem to have a consensus: opm-models!

@akva2 can you take charge of the process? It seems to me this is your domain mostly.

**Comment by atgeirr** [comment]
This seems to have worked out well, closing!

**Comment by atgeirr** [comment]
I'll just add a list of things I did locally as a developer to adjust to this change, in case it is of use to others.

1. Rename (ewoms->opm-models) your own github fork of the repository.
2. Rename your local checked-out source code directory.
3. Ensure git points to the correct repositories. First "git remote -v" in the module to see what you have, then "git remote set-url ..." to point to the renamed repos, finally "git remote update" to verify you did it correctly.
4. If you use any build scripts or similar, edit them.
5. Rename and clean out the build folder(s), run cmake again for opm-models.
6. Run a full build to ensure all is in good shape.
